pkg/resolver/client/ens/ens_integration_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build integration
+
+package ens_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/resolver/client/ens"
+)
+
+func TestENSntegration(t *testing.T) {
+	// TODO: consider using a stable gateway instead of INFURA.
+	defaultEndpoint := "https://goerli.infura.io/v3/59d83a5a4be74f86b9851190c802297b"
+
+	testCases := []struct {
+		desc            string
+		endpoint        string
+		name            string
+		wantAdr         string
+		wantFailConnect bool
+		wantFailResolve bool
+	}{
+		{
+			desc:            "bad ethclient endpoint",
+			endpoint:        "fail",
+			wantFailConnect: true,
+		},
+		{
+			desc:            "no domain",
+			name:            "idonthaveadomain",
+			wantFailResolve: true,
+		},
+		{
+			desc:            "no eth domain",
+			name:            "centralized.com",
+			wantFailResolve: true,
+		},
+		{
+			desc:            "not registered",
+			name:            "unused.test.swarm.eth",
+			wantFailResolve: true,
+		},
+		{
+			desc:            "no content hash",
+			name:            "nocontent.resolver.test.swarm.eth",
+			wantFailResolve: true,
+		},
+		{
+			desc:    "ok",
+			name:    "example.resolver.test.swarm.eth",
+			wantAdr: "00cb23598c2e520b6a6aae3ddc94fed4435a2909690bdd709bf9d9e7c2aadfad",
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			if tC.endpoint == "" {
+				tC.endpoint = defaultEndpoint
+			}
+
+			eC := ens.NewClient()
```

Comment:
Defer a function that will close the client if it is still connected, as it will consume resources in case of an error in between the explicit Close call.